app/views/trial_scorings/show.html.haml

```diff
@@ -17,6 +17,10 @@
       %tr
         %th.plant_scoring_units_scoring_unit_name_column
           = t 'tables.plant_scoring_units.scoring_unit_name'
-        - @trait_descriptors.map{ |td| td.try(:trait).try(:name) }.each do |column|
-          %th= column
+        - @trait_descriptors.map{ |td| [td.id, td.try(:trait).try(:name)] }.each do |id, column|
```

Comment:
I think not. Both have `null: false` in the schema and there is the FK constraint on the `TD.trait_id` column, so that should not happen. In worst case scenario, T.name could be an empty string.